### PR TITLE
Feature/ISBD 257 - Cookies Preferences 

### PR DESCRIPTION
**Pull request by hmcts-abduldaim**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ISDB-257

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by yogendraU — APPROVED** [review]

**Review by rahulsanjava19 — APPROVED** [review]

File: src/main/app/controller/GetController.ts [hunk 1]
```diff
@@ -43,34 +61,84 @@ export class GetController {
...
+    /**
+     *
+     *                      This util allows to delete document
+     *                      the params it uses is @ds-web-cookie-preferences
+     *                      This is used to check for current cookiesPreferences
+     *      ************************************ ************************************
+     *      ************************************  ************************************
+     *
+     */
+    let PageRenderableContents = {
```
Comment (parvezhmcts) [inline src/main/app/controller/GetController.ts:112]: variable name should in camelCase.

File: src/main/app/controller/GetController.ts [hunk 2]
```diff
@@ -43,34 +61,84 @@ export class GetController {
...
+    /**
+     *
+     *                      This util allows saved cookies to have redirect after successfully saving
+     *                      the params it uses is @ds-web-cookie-preferences
+     *                      This is used to check for current cookiesPreferences
+     *      ************************************ ************************************
+     *      ************************************  ************************************
+     *
+     */
+    const CookieWithSaveQuery = COOKIES + '?togglesaveCookie=true';
```
Comment (parvezhmcts) [inline src/main/app/controller/GetController.ts:133]: should be in camelCase.

File: src/main/app/controller/GetController.ts [hunk 3]
```diff
@@ -124,6 +192,59 @@ export class GetController {
...
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      cookieExpiryDuration = cookieExpiryDuration * TimeInADay; //cookie time in milliseconds
+      const CookiePreferences = {
+        analytics: '',
+        apm: '',
+      };
+      if (req.query.hasOwnProperty('analytics')) {
+        switch (req.query['analytics']) {
+          case 'off':
+            CookiePreferences['analytics'] = 'off';
```
Comment (parvezhmcts) [inline src/main/app/controller/GetController.ts:211]: for 'off' and 'on' creating constant and using it will be good for maintainability perspective.

**Review by parvezhmcts — CHANGES_REQUESTED** [review]

File: src/main/app/controller/GetController.ts (hunk 3 above)
Comment (hmcts-abduldaim) [reply src/main/app/controller/GetController.ts]: has been resolved

**Review by parvezhmcts — APPROVED** [review]

**Pull request merged**